pkg/provider/redirect.go

```diff
@@ -81,14 +83,28 @@ func createRedirectSignature(
 		return err
 	}
 
-	sigAlg := idp.signingContext.GetSignatureMethodIdentifier()
-	sig, err := signature.CreateRedirect(idp.signingContext, buildRedirectQuery(string(respData), response.RelayState, sigAlg, ""))
+	cert, key, err := getResponseCert(ctx, idp.storage)
+	if err != nil {
+		return err
+	}
+
+	tlsCert, err := signature.ParseTlsKeyPair(cert, key)
+	if err != nil {
+		return err
+	}
+
+	signingContext, err := signature.GetSigningContext(tlsCert, idp.conf.SignatureAlgorithm)
+	if err != nil {
+		return err
+	}
+
+	sig, err := signature.CreateRedirect(signingContext, buildRedirectQuery(string(respData), response.RelayState, idp.conf.SignatureAlgorithm, ""))
 	if err != nil {
 		return err
 	}
 
 	response.Signature = url.QueryEscape(base64.StdEncoding.EncodeToString(sig))
-	response.SigAlg = url.QueryEscape(base64.StdEncoding.EncodeToString([]byte(sigAlg)))
+	response.SigAlg = url.QueryEscape(base64.StdEncoding.EncodeToString([]byte(idp.conf.SignatureAlgorithm)))
```

Comment:
why do you need to url encode after base64 encoding?

Reply:
to cover everything clearly, for example, if you refactor, by redirect-binding everything is in the URL